engine/runtime/src/main/scala/org/enso/compiler/pass/optimise/LambdaConsolidate.scala

```diff
@@ -319,6 +319,7 @@ case object LambdaConsolidate extends IRPass {
             }
           )
         case self: IR.Name.Self             => self
+        case self: IR.Name.SelfType         => self
```

Comment:
```suggestion
        case selfType: IR.Name.SelfType         => selfType
```

Okay this is extremely pedantic (I'm sorry :smiling_face_with_tear: ), but IMO it looks better if the two look distinct - otherwise my first look at this code was "Hey, why are these two the same, these are two different things after all. OH they are just sharing the name and that's all, ok." - not a big deal but we can avoid the first confusion part and go to the second one by changing the name.